tests/foreman/destructive/test_capsulecontent.py

```diff
@@ -91,3 +92,97 @@ def test_positive_sync_without_deadlock(
+        1. Blank external capsule which is not synced yet, running with 4 & more pulpcore-workers
+
+    :steps:
+        1. Sync one bigger repositories to the Satellite.
+        2. Create a Content View, add the repository and publish it.
+        3. Promote it to 10+ LCEs and assign them to Capsule
+        4. Sync the Capsule
+        5. pulpcore-manager rpm-trim-changelogs --changelog-limit x
+        6. Sync Capsule again and verify no deadlocks occurs.
+        7. Repeat step 5 with a lesser changelog limit and step 6 again
+
+    :expectedresults:
+        1. Sync passes without deadlock.
+
+    :customerscenario: true
+
+    :BZ: 2170535, 2218661
+    """
+    org = function_entitlement_manifest_org
+    rh_repos = []
+    tasks = []
+    LCE_COUNT = 10
+    for name in ['rhel8_bos', 'rhel8_aps']:
+        rh_repo_id = target_sat.api_factory.enable_rhrepo_and_fetchid(
+            basearch=constants.DEFAULT_ARCHITECTURE,
+            org_id=org.id,
+            product=constants.REPOS[name]['product'],
+            repo=constants.REPOS[name]['name'],
+            reposet=constants.REPOS[name]['reposet'],
+            releasever=constants.REPOS[name]['releasever'],
+        )
+        # Sync step because repo is not synced by default
+        rh_repo = target_sat.api.Repository(id=rh_repo_id).read()
+        task = rh_repo.sync(synchronous=False)
+        tasks.append(task)
+        rh_repos.append(rh_repo)
+    for task in tasks:
+        target_sat.wait_for_tasks(
+            search_query=(f'id = {task["id"]}'),
+            poll_timeout=2500,
+        )
+        task_status = target_sat.api.ForemanTask(id=task['id']).poll()
+        assert task_status['result'] == 'success'
+
+    cv = target_sat.publish_content_view(org, rh_repos)
+    cvv = cv.version[0].read()
+
+    for i in range(LCE_COUNT):
+        lce = target_sat.api.LifecycleEnvironment(name=f'lce{i}', organization=org).create()
+        cvv.promote(data={'environment_ids': lce.id, 'force': False})
+        capsule_configured.nailgun_capsule.content_add_lifecycle_environment(
+            data={'environment_id': lce.id}
+        )
+
+    result = Box(capsule_configured.nailgun_capsule.content_lifecycle_environments())
+    assert len(result['results']) == LCE_COUNT
+    # Verify all LCE have a CV promoted
+    assert [env.content_views[0].name for env in result.results].count(cv.name) == LCE_COUNT
+
+    sync_status = capsule_configured.nailgun_capsule.content_sync(timeout='60m')
+    assert sync_status['result'] == 'success', 'Capsule sync task failed'
+
+    # Run rpm-trim-changelogs with changelog-limit below 10 then again sync capsule
+    # and repeat again with changelog-limit below it to make sure deadlock
+    pulp_cmd = (
+        'sudo -u pulp PULP_SETTINGS="/etc/pulp/settings.py" '
+        '/usr/bin/pulpcore-manager rpm-trim-changelogs --changelog-limit '
+    )
+    for limit in range(9, 0, -1):
+        result = target_sat.execute(pulp_cmd + str(limit))
```

Comment:
non-blocking: I see f-string as a better way then string concatenation. 
```suggestion
        result = target_sat.execute(f'{pulp_cmd}{limit}')
```